src/transmission.js

```diff
@@ -284,79 +283,86 @@ export class Transmission {
     let batches = Object.keys(batchAgg.batches).map(k => batchAgg.batches[k]);
     eachPromise(batches, batch => {
       let url = urljoin(batch.apiHost, "/1/batch", batch.dataset);
-      let req = superagent.post(url);
-      if (this._proxy) {
-        req = proxy(req, this._proxy);
-      }
+      let postReq = superagent.post(url);
 
-      let { encoded, numEncoded } = batchAgg.encodeBatchEvents(batch.events);
-      return new Promise(resolve => {
-        // if we failed to encode any of the events, no point in sending anything to honeycomb
-        if (numEncoded === 0) {
-          this._responseCallback(
-            batch.events.map(ev => ({
-              metadata: ev.metadata,
-              error: ev.encodeError
+      const reqPromise = Promise.resolve(
+        this._proxy
+          ? import("superagent-proxy").then(({ default: proxy }) => ({
+              req: proxy(postReq, this._proxy)
             }))
-          );
-          resolve();
-          return;
-        }
-
-        let userAgent = USER_AGENT;
-        let trimmedAddition = this._userAgentAddition.trim();
-        if (trimmedAddition) {
-          userAgent = `${USER_AGENT} ${trimmedAddition}`;
-        }
-
-        let start = Date.now();
-        req
-          .set("X-Honeycomb-Team", batch.writeKey)
-          .set("User-Agent", userAgent)
-          .type("json")
-          .send(encoded)
-          .end((err, res) => {
-            let end = Date.now();
-
-            if (err) {
+          : { req: postReq }
```

Comment:
this object is necessary because superagent requests are thenable, so if we return the superagent request directly from either the proxy then on line 290, or here, `Promise.resolve` will call the `.then` method on the request object, causing the request to actually fire.  🤦‍♂